### PR TITLE
Fix bug where node adaption failed for nodes with repeated inputs

### PR DESCRIPTION
**Pull request by cbourjau**
Node-adaption currently fails if the adapted node has repeating inputs (e.g. the result of `op.equal(a, a)`). This PR addresses that issue.

# Checklist

- [x] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Review by jbachurski — APPROVED** [review]
LGTM. 
Feel free to have a second review.
This logic has always been tricky and refactoring it might be worth it at some point 😃

**Comment by cbourjau** [comment]
Thanks for the quick review! I ran into another similar issue where we need to adapt nodes in subgraphs. I'll make a separate PR for that next week. This code path is now getting some real-world use since the release of opset version 19.

**Pull request merged**
